File: app/src/main/java/com/example/ideal/myapplication/adapters/DialogAdapter.java
package com.example.ideal.myapplication.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.ideal.myapplication.R;
import com.example.ideal.myapplication.adapters.chatElements.DialogElement;
import com.example.ideal.myapplication.fragments.objects.Dialog;
import com.example.ideal.myapplication.fragments.objects.Message;
import com.example.ideal.myapplication.other.DBHelper;

import java.util.ArrayList;

public class DialogAdapter extends RecyclerView.Adapter<DialogAdapter.DialogViewHolder> {

    private int numberItems;
    private ArrayList<Dialog> dialogList;
    private Context context;
    private DBHelper dbHelper;
    private static final String TAG = "DBInf";

    public DialogAdapter(int numberItems, ArrayList<Dialog> dialogList, DBHelper dbHelper) {
        this.numberItems = numberItems;
        this.dialogList = dialogList;
        this.dbHelper = dbHelper;
    }

    @NonNull
    @Override
    public DialogViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        context = viewGroup.getContext();
        int layoutIdForListItem = R.layout.dialog_element;
        //Класс, который позволяет создавать представления из xml файла
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        // откуда, куда, необходимо ли помещать в родителя
        View view = layoutInflater.inflate(layoutIdForListItem, viewGroup, false);

        DialogViewHolder viewHolder = new DialogViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull DialogViewHolder dialogViewHolder, int i) {
        dialogViewHolder.bind(dialogList.get(i));
    }

    @Override
    public int getItemCount() {
        return numberItems;
    }

    class DialogViewHolder extends RecyclerView.ViewHolder {

        View view;

        DialogViewHolder(@NonNull View itemView) {
            super(itemView);
            view = itemView;
        }

        void bind(Dialog dialog) {
            DialogElement dialogElement = new DialogElement(dialog,view,context, dbHelper);
        }
    }
}

File: app/src/main/java/com/bunbeauty/ideal/myapplication/clean_architecture/adapters/DialogAdapter.kt
package com.bunbeauty.ideal.myapplication.clean_architecture.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.android.ideal.myapplication.R
import com.bunbeauty.ideal.myapplication.clean_architecture.adapters.elements.chatElements.DialogElement
import com.bunbeauty.ideal.myapplication.clean_architecture.data.db.models.entity.Dialog
import com.bunbeauty.ideal.myapplication.clean_architecture.data.db.models.entity.Message

class DialogAdapter : RecyclerView.Adapter<DialogAdapter.DialogViewHolder>() {

    private val dialogList: ArrayList<Dialog> = arrayListOf()

    fun addItem(dialog: Dialog) {
        val foundDialog = dialogList.find { it.user.id == dialog.user.id }
        if (foundDialog == null) {
            dialogList.add(dialog)
            dialogList.sortByDescending { it.lastMessage.time }
            notifyItemInserted(dialogList.size)
        } else {
            val index = dialogList.indexOf(foundDialog)
            dialogList[index] = dialog
            dialogList.sortByDescending { it.lastMessage.time }
            notifyItemChanged(index)
        }
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): DialogViewHolder {
        val context = viewGroup.context
        val layoutIdForListItem = R.layout.element_dialog
        val layoutInflater = LayoutInflater.from(context)
        val view = layoutInflater.inflate(layoutIdForListItem, viewGroup, false)
        return DialogViewHolder(view, context)
    }

    override fun onBindViewHolder(dialogViewHolder: DialogViewHolder, i: Int) {
        dialogViewHolder.bind(dialogList[i])
    }

    override fun getItemCount() = dialogList.size

    inner class DialogViewHolder(private val view: View, private val context: Context) :
        ViewHolder(view) {
        fun bind(dialog: Dialog) {
           DialogElement(view, context, dialog)
        }
    }


}